geode-core/src/main/java/org/apache/geode/internal/ObjectInputStreamFilterWrapper.java

```diff
@@ -0,0 +1,100 @@
+
+import org.apache.logging.log4j.Logger;
+import sun.misc.ObjectInputFilter;
+
+import org.apache.geode.InternalGemFireException;
+import org.apache.geode.distributed.internal.DistributedSystemService;
+import org.apache.geode.internal.logging.LogService;
+
+
+public class ObjectInputStreamFilterWrapper implements InputStreamFilter {
+  private static final Logger logger = LogService.getLogger();
+  private final ObjectInputFilter serializationFilter;
+
+  public ObjectInputStreamFilterWrapper(String serializationFilterSpec,
+      Collection<DistributedSystemService> services) {
+
+    Set<String> sanctionedClasses = new HashSet<>(500);
+    for (DistributedSystemService service : services) {
+      try {
+        Collection<String> classNames = service.getSerializationWhitelist();
+        logger.info("loaded {} sanctioned serializables from {}", classNames.size(),
+            service.getClass().getSimpleName());
+        sanctionedClasses.addAll(classNames);
+      } catch (IOException e) {
+        throw new InternalGemFireException("error initializing serialization filter for " + service,
+            e);
+      }
+    }
+
+    try {
+      URL sanctionedSerializables = ClassPathLoader.getLatest()
+          .getResource(InternalDataSerializer.class, "sanctioned-geode-core-serializables.txt");
+      Collection<String> coreClassNames =
+          InternalDataSerializer.loadClassNames(sanctionedSerializables);
+      sanctionedClasses.addAll(coreClassNames);
+    } catch (IOException e) {
+      throw new InternalGemFireException(
+          "unable to read sanctionedSerializables.txt to form a serialization white-list", e);
+    }
+
+    logger.info("setting a serialization filter containing {}", serializationFilterSpec);
+    // logger.info("sanctioned serializables are");
+    // for (String cls: sanctionedClasses) {
+    // logger.info(" " + cls);
+    // }
+
+    final ObjectInputFilter userFilter =
+        ObjectInputFilter.Config.createFilter(serializationFilterSpec);
+    serializationFilter = filterInfo -> {
+      if (filterInfo.serialClass() == null) {
+        return userFilter.checkInput(filterInfo);
+      }
+
+      String className = filterInfo.serialClass().getName();
+      if (filterInfo.serialClass().isArray()) {
+        className = filterInfo.serialClass().getComponentType().getName();
+      }
+      if (sanctionedClasses.contains(className)) {
+        return ObjectInputFilter.Status.ALLOWED;
+        // return ObjectInputFilter.Status.UNDECIDED;
+      } else {
+        ObjectInputFilter.Status status = userFilter.checkInput(filterInfo);
+        if (status == ObjectInputFilter.Status.REJECTED) {
+          logger.fatal("Serialization filter is rejecting class {}", className, new Exception(""));
+        }
+        return status;
+      }
+    };
+
+    // global filter - if we enable this it will affect all ObjectInputStreams
+    // ObjectInputFilter.Config.setSerialFilter(serializationFilter);
```

Comment:
Also, did you want to remove this before pushing?

Reply:
ditto